nuxeo-java-client-test/src/test/java/org/nuxeo/client/ITDirectory.java

```diff
@@ -100,4 +102,33 @@ public void itCanCreateUpdateFetchDeleteDirectory() {
         directory.deleteEntry(result.getId());
     }
 
+    @Test
+    public void itCanSetOrderingPropertyType() {
+        Directory directory = nuxeoClient.directoryManager().directory("continent");
+
+        DirectoryEntry entryWithInteger = new DirectoryEntry();
+        DirectoryEntry entryWithLong = new DirectoryEntry();
+        entryWithInteger.setDirectoryName("continent");
+        entryWithInteger.putIdProperty("test1");
+        entryWithInteger.putLabelProperty("test1");
+        entryWithInteger.putObsoleteProperty(0);
+
+        entryWithLong.setDirectoryName("continent");
+        entryWithLong.putIdProperty("test2");
+        entryWithLong.putLabelProperty("test2");
+        entryWithLong.putObsoleteProperty(1);
+
+        entryWithLong.putOrderingProperty(Long.MAX_VALUE);
+        entryWithInteger.putOrderingProperty(Integer.valueOf(Integer.MAX_VALUE).longValue());
+        DirectoryEntry resultWithInteger = directory.createEntry(entryWithInteger);
+        DirectoryEntry resultWithLong = directory.createEntry(entryWithLong);
+        assertNotNull(resultWithInteger);
+        assertNotNull(resultWithLong);
+
+        resultWithInteger = resultWithInteger.update();
```

Comment:
After the spit, in the Integer way test, could you add an extra test that put a number in the Long-range, assert the putOrderingProperty returned object, update the entry, and then assert the getOrderingProperty object?